### PR TITLE
Mod picker move

### PR DESCRIPTION
**Pull request by ryan-rushton**
Ok so this is a bit of house keeping so `loadout-builder` and `the folder formally known as loadout` didn't depend on each other (I have run into a few cyclic deps lately). 

Basically I have done the following 
1. Renamed the `loadout` folder to `loadout-drawer`
2. Moved the ModPicker to a new folder which I called `loadout`
3. Moved ModPicker's dependencies into the new `loadout` folder too, note that this includes splitting up the types, mod-utils, and utils files in `loadout-builder`
4. Renamed isLoadoutBuilderItem function to isArmor2WIthStats as it made more sense now it's not in the `loadout-builder` dir

### CODE REVIEW

File: src/app/loadout/mod-utils.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
+import { PluggableInventoryItemDefinition } from 'app/inventory/item-types';
```
Comment (ryan-rushton) [inline src/app/loadout/mod-utils.ts:1]: No changes in this file, its a straight copy paste

File: src/app/loadout/known-values.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import {
+  armor2PlugCategoryHashes,
+  armor2PlugCategoryHashesByName,
+  D2ArmorStatHashByName,
+} from 'app/search/d2-known-values';
+import { PlugCategoryHashes } from 'data/d2/generated-enums';
+
+export const armorStatHashes = [
```
Comment (ryan-rushton) [inline src/app/loadout/known-values.ts:8]: This is a replacement for the statValues array in loadout-builder/types

File: src/app/loadout/item-utils.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { DimItem } from 'app/inventory/item-types';
+import { armorStatHashes } from './known-values';
+
+/** Whether this item is eligible for being in loadout builder. Now only armour 2.0 and only items that have all the stats. */
+
+export function isArmor2WithStats(item: DimItem) {
```
Comment (ryan-rushton) [inline src/app/loadout/item-utils.ts:6]: This is `isLoadoutBuilderItem` but I have renamed and replaced `statValues` with `armorStatHashes` (they are both arrays of the 6 armor stat hashes.)

File: src/app/loadout/item-utils.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import { DimItem } from 'app/inventory/item-types';
+import { armorStatHashes } from './known-values';
+
+/** Checks if the item is Armor 2.0 and whether it has stats present for all 6 armor stats. */
+export function isArmor2WithStats(item: DimItem) {
```
Comment (ryan-rushton) [inline src/app/loadout/item-utils.ts:5]: This is isLoadoutBuilderItem but I have renamed and replaced statValues with armorStatHashes (they are both arrays of the 6 armor stat hashes.)

**Comment by ryan-rushton** [comment]
@bhollis and @nev-r, planning on merging this tomorrow so I can move on to some other work unless either of you have objections.

**Comment by bhollis** [comment]
sounds good!

**Pull request merged**